From 70d4fcf586ea315b33df8bac3da8f5f61baf1fb8 Mon Sep 17 00:00:00 2001
From: Botond Kardos <bkardos70@gmail.com>
Date: Wed, 27 Apr 2022 17:26:50 +0200
Subject: [PATCH] console-server now tolerates virtual UART devices

---
 console-server.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/console-server.c b/console-server.c
index 11d017b..3d18689 100644
--- a/console-server.c
+++ b/console-server.c
@@ -138,9 +138,14 @@ static int tty_find_device(struct console *console)
 	if (rc < 0)
 		goto out_free;
 
+	rc = asprintf(&console->tty_dev, "%s", tty_path_input_real);
+	if (rc < 0)
+		goto out_free;
+
 	tty_device_tty_dir = realpath(tty_class_device_link, NULL);
 	if (!tty_device_tty_dir) {
 		warn("Can't query sysfs for device %s", tty_kname_real);
+		rc = 0;
 		goto out_free;
 	}
 
@@ -152,10 +157,6 @@ static int tty_find_device(struct console *console)
 	if (!console->tty_sysfs_devnode)
 		warn("Can't find parent device for %s", tty_kname_real);
 
-	rc = asprintf(&console->tty_dev, "/dev/%s", tty_kname_real);
-	if (rc < 0)
-		goto out_free;
-
 	rc = 0;
 
 out_free:
@@ -174,6 +175,9 @@ static int tty_set_sysfs_attr(struct console *console, const char *name,
 	FILE *fp;
 	int rc;
 
+	if (!console->tty_sysfs_devnode)
+		return 0;
+
 	rc = asprintf(&path, "%s/%s", console->tty_sysfs_devnode, name);
 	if (rc < 0)
 		return -1;
@@ -842,7 +846,8 @@ out_config_fini:
 out_free:
 	free(console->pollers);
 	free(console->pollfds);
-	free(console->tty_sysfs_devnode);
+	if (console->tty_sysfs_devnode)
+		free(console->tty_sysfs_devnode);
 	free(console->tty_dev);
 	free(console);
 
-- 
2.25.1

